index.js

```diff
@@ -20,75 +16,123 @@ let defaults = {
-     return encodeURIComponent(param);
+    log(method, 'info');
+  });
+
+  // call this to extra params from request by GET or POST:
+  // note: I think we're mainly wanting to support JSON/POST requests
+  function extractParamsFromRequest(request){
+    console.log(request.method)
+    if (request.method.toLowerCase()=='post'){
+      log("using POST/JSON method ")
+      log(request.payload)
+      var params = _.map(request.payload.values, function(param){
+        return decodeURIComponent(param);
+      });
+      return params;
+    }
+    else return extractParamsFromRequestGET(request);
+  }
+  // helper that is called by extractParamsFromRequest for GET requests:
+  // if you pass in a JSON package with 'values' it uses that
+  // otherwise it tries to extract params directly from the url path:
+  function extractParamsFromRequestGET(request){
+   log("using GET/url params method")
+   // gets empty array if params is undefined
+   var params = (!request.params.params) ? [] : _.map(request.params.params.split("/"), function(param){
+     return decodeURIComponent(param);
    });
-   params.push(done);
    return params;
   }
-  // probably need something more robust but this works for 1 level:
-  function getMethodFromServerWithNamespace(server, methodName){
-    if (methodName.indexOf(".")>-1){
-      var fields = methodName.split(".");
-      return server.methods[_.first(fields)][_.last(fields)];
-    }
-    return server.methods[methodName];
+
+  // helper functions to reply to a request,
+  // depending on the preferred return method
+  function replyAsHTML( failureCode, response, reply){
+    if (failureCode)
+      reply(`<h1>${failureCode}</h1> <br> ${response}`).code(failureCode);
+    else
+      reply(`<h1>Successful!<h1> <br> Result is: ${response}`);
   }
+  function replyAsJSON(failureCode, response, reply){
+    if (failureCode)
+      reply({statusCode:failureCode, error: "Method Call Failed", message: response}).code(failureCode);
+    else
+      reply({successful: true, result: response})
+  }
+  // change this to switch the reply method from JSON to HTML
+  // by default we're responding with JSON:
+  var replyHandler = replyAsJSON; // replyAsHTML;
 
   log("methodsRoutePlugin is setting up a route at : " + endpoint);
   server.route({
-      method: 'GET',
+      method: '*',
       path: endpoint + '/{methodName}/{params*}',
       handler: function (request, reply) {
-        let methodName = encodeURIComponent(request.params.methodName);
-        if (!blocklist(methodName, request)){
-          reply(`<h1> 403 Error </h1> <br> You do not have access to ${methodName} `).code(403)
+        // get the method they are trying to call:
+        var methodName = decodeURIComponent(request.params.methodName);
+        // this goes at the top because we want to short-circuit if they don't have access
+        // to the indicated method:
+        if (!whitelist(methodName, request)){
```

Comment:
maybe whitelist should just be an array instead of a function?  thoughts on that?